provider/datagen/src/baked_exporter.rs

```diff
@@ -525,17 +525,6 @@ impl BakedExporter {
                         .insert("icu_locid_transform/compiled_data");
                     let search_direct = search(quote!(req.locale));
                     let search_iterator = search(quote!(fallback_iterator.get()));
-                    let maybe_err = if keys.contains(&String::from("und")) {
-                        // The loop will terminate on its own
```

Comment:
In fact, adding auxiliary keys breaks the invariant that any `DataLocale` will eventually fall back to `und`. Even worse, *users* can break this invariant because we have `DataLocale` on all our constructors instead of having it as an internal type. This is something we need to consider before stabilizing.

Reply:
Users can break the invariant whether or not we have DataLocale on our constructors by invoking DataProvider directly. It's trivial to trigger an infinite loop today by using aux keys.

I don't see how making the function have a `break` changes the inlining properties of the function.

I'm not excited about making BakedExporter have special logic about aux keys. It should treat locales as opaque except for calling `.is_und()` to terminate the loop.

The UTS 35 locale fallback algorithm concerns itself only with language identifiers, and we still have the invariant that the language identifier eventually falls back to `und`. ICU4X additionally has the invariant that Unicode extension keywords eventually are removed. I will open a new PR documenting these invariants.

Reply:
I can see a future where the DataKeyMetadata specifies behavior regarding aux key fallback for that specific DataKey, but I do think the default behavior should be to not fall back on the aux key.

Reply:
Yes, and this PR makes it impossible on experimental, too.